### PR TITLE
add shellcheck test, fix code for shellcheck

### PR DESCRIPTION
**Pull request by ogra1**
- sets up xenial env
- runs shellcheck
- builds two source packages
- builds two binary packages

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+language: C
+dist: xenial
```
Comment (zyga) [inline .travis.yml]: Is that a real thing now? I thought travis just "upgraded" defaults to 14.04 but only supported precise and trusty.

File: config/usr/bin/mkswapfile [hunk 2]
```diff
@@ -11,35 +11,35 @@ set -e
...
 if [ ! -e "$FILE" ]; then
-  create_swapfile $FILE $SIZE
+  create_swapfile "$FILE" "$SIZE"
 fi
 
 # file exists but has the wrong size
-if [ "$(stat -c %s $FILE)" -ne "$((1024*1024*$SIZE))" ]; then
-  delete_swapfile $FILE
-  create_swapfile $FILE $SIZE
+if [ "$(stat -c %s "$FILE")" -ne "$((1024*1024*SIZE))" ]; then
```
Comment (zyga) [inline config/usr/bin/mkswapfile]: Wow that's some pretty weird syntax.

File: initramfs/scripts/local-premount/resize [hunk 3]
```diff
@@ -31,24 +31,24 @@ done
...
 {
     NUM=$1
     lastpart="$(grep -cE ^'[0-9]{1,}': $TMPFILE)"
     if [ "$lastpart" = "$NUM" ]; then
-        endsize="$(parted -ms $DEV print| grep ^/ | cut -d: -f2)"
+        endsize="$(parted -ms "$DEV" print| grep ^/ | cut -d: -f2)"
     else
         # we are not at the end ! get the start of the next partition
         # (minus 1 byte) instead of using the absolute end of the disk
-        endsize=$(($(parted -ms $DEV unit B print|grep ^$(($num+1)):|\
```
Comment (zyga) [inline initramfs/scripts/local-premount/resize]: So `$num` was incorrect here? Just checking

File: .travis.yml (hunk 1 above)
Comment (ogra1) [inline .travis.yml]: well, it isnt ... but it is also ignored and describes that we now use a xenial chroot inside when testing ;)

File: config/usr/bin/mkswapfile (hunk 2 above)
Comment (ogra1) [inline config/usr/bin/mkswapfile]: it is comparing byte sizes ... which part do you find weird

File: initramfs/scripts/local-premount/resize (hunk 3 above)
Comment (ogra1) [inline initramfs/scripts/local-premount/resize]: yes, $num never existed, and likewise we do not have any image yet where the "else" would actually be in effect (only if "writable" is not the last partition on disk, we do not have such images yet)

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  - sudo mount --bind /dev $CHROOT/dev
+  - sudo chroot $CHROOT apt update
+  - sudo chroot $CHROOT apt install -y shellcheck file build-essential devscripts
+  - sudo cp -a initramfs config $CHROOT/build/
+    # install all build deps defined in the control file
+  - sudo chroot $CHROOT apt install -y $(grep -h Build-Depends {initramfs,config}/debian/control|sed 's/^.*://g;s/,//g;s/([^)]*)//g'| tr -d '\n'| sort | uniq)
+script:
+  - sudo chroot $CHROOT sh -c 'cd build; for file in $(find . | xargs file | grep shell |grep -v .git| sed "s/:.*$/ /g" | tr -d "\n"); do shellcheck $SHCKOPTS $file; done'
+    # test build src packages unsigned in a xenial chroot
+  - sudo chroot xenial-test-chroot sh -c 'cd build/initramfs; LC_ALL=C.UTF-8 dpkg-buildpackage -rfakeroot -S -sa -us -uc'
```
Comment (niemeyer) [inline .travis.yml]: Why above uses `$CHROOT` but these don't?

The common prefix here might also be turned into a variable btw (`$CHROOT_RUN` or something).

Comment (ogra1) [inline .travis.yml]: nice catch! i was testing the line locally and forgot to substitute the variable back in when i copy/pasted it after testing...

File: initramfs/scripts/local-premount/resize [hunk 5]
```diff
@@ -58,55 +58,55 @@ do_gpt()
...
+    sgdisk -p "$DISK" >/run/initramfs/new-gpt-table.txt 2>/dev/null
 }
 
 do_mbr()
 {
     DEV=$1
     PART=$2
-    endsize=$(get_end $PART)
-    parted -s $DEV resizepart $PART $endsize
+    endsize=$(get_end "$PART")
```
Comment (zyga) [inline initramfs/scripts/local-premount/resize]: You can also quote the outer part but I think it's actually not required here.

**Pull request merged**